uhabits-core/src/jvmMain/java/org/isoron/uhabits/core/utils/DateUtils.kt

```diff
@@ -178,6 +179,26 @@ abstract class DateUtils {
             return getWeekdayNames(GregorianCalendar.SHORT, firstWeekday)
         }
 
+        /**
+         * Returns a vector of Int representing the frequency of each weekday in a given month.
+         *
+         * @param startOfMonth a Timestamp representing the beginning of the month.
+         */
+        @JvmStatic
+        fun getWeekdaysInMonth(startOfMonth: Timestamp): Array<Int> {
+            val month = startOfMonth.toCalendar()[Calendar.MONTH] + 1
+            val year = startOfMonth.toCalendar()[Calendar.YEAR]
+            val weekday = startOfMonth.weekday
+            val extraWeekdays = YearMonth.of(year, month).lengthOfMonth() - 28
+
+            val freq = Array(7) { 4 }
+            for (day in weekday until weekday + extraWeekdays) {
+                freq[day % 7] = 5
+            }
```

Comment:
I think it might more readable if you just iterate through all the days of the month and add +1 to the corresponding entry in the array at every iteration.

Reply:
Yeah I see what you mean, I went with this approach to reduce the time complexity of the algorithm, but I can see how it may be confusing to read.

Reply:
Changed it to the method you described.